cranelift/isle/isle/src/files.rs

```diff
@@ -109,6 +121,15 @@ impl Files {
     }
 }
 
+/// If OUT_DIR is set, strips it from the file name.
+/// Otherwise returns the original file name.
+pub fn relative(file_name: &str) -> &str {
+    match std::env::var("OUT_DIR") {
+        Err(_) => file_name,
+        Ok(root) => file_name.strip_prefix(&root).unwrap_or(file_name).into(),
+    }
+}
```

Comment:
I wonder if the build script that invokes cranelift-isle can read this instead. Cranelift-isle can technically be used to build arbitrary code. `Files::from_paths` could get a set of paths to make make input paths relative to, or alternatively a set of paths against which to resolve relative paths and then have the build script only pass the relative paths for individual isle files instead.

Reply:
Sorry, not too familiar with the code/cranelift. Are you suggesting that we should make the paths relative [here](https://github.com/bytecodealliance/wasmtime/blob/06377eb08a649619cc8ac9a934cb3f119017f3ef/cranelift/codegen/build.rs#L225) instead?

Reply:
Yes. There is already a `make_isle_source_path_relative` function which gets used, but that only affects isle files that are part of the source rather than those that get generated by the build script itself. You did still have to separately pass a (set of) prefix(s) to join with the relative path to get a path that can be passed to `fs::read`, but this prefix can be ignored when printing the source locations in the generated file.

Reply:
>  only affects isle files that are part of the source rather than those that get generated by the build script itself. 

Unrelated to this thread, but that explains why I had to use `OUT_DIR` instead of `CARGO_MANIFEST_DIR` !

> Yes. 

That makes sense. I'll try to find some time later to update the PR.